numba_dpex/dpnp_iface/arrayobj.py

```diff
@@ -1078,14 +1082,29 @@ def getitem_arraynd_intp(context, builder, sig, args):
     that when returning a view of a dpnp.ndarray the sycl::queue pointer
     member in the LLVM IR struct gets properly updated.
     """
-    ret = np_getitem_arraynd_intp(context, builder, sig, args)
+    is_kernel = isinstance(context, DpexKernelTargetContext)
```

Comment:
A better name is needed. How about `getitem_call_in_kernel`?

Reply:
Whatever sounds better here. I don't have preferences. This function is really short. Ideally it won't be an issue with target specific overload.